lldb/source/Plugins/TypeSystem/Swift/TypeSystemSwiftTypeRef.cpp

```diff
@@ -1907,20 +1940,125 @@ CompilerType TypeSystemSwiftTypeRef::GetFieldAtIndex(
-      child_is_base_class, child_is_deref_of_parent, valobj, language_flags);
+    ValueObject *valobj, uint64_t &language_flags)
+
+  __attribute__((optnone))
+
+
+
+{
+  auto impl = [&]() -> CompilerType {
+    ExecutionContextScope *exe_scope = nullptr;
+    if (exe_ctx)
+      exe_scope = exe_ctx->GetBestExecutionContextScope();
+    if (exe_scope) {
+      if (auto *runtime =
+            SwiftLanguageRuntime::Get(exe_scope->CalculateProcess()))
+        if (CompilerType result = runtime->GetChildCompilerTypeAtIndex(
+                {this, type}, idx, transparent_pointers,
+                omit_empty_base_classes, ignore_array_bounds, child_name,
+                child_byte_size, child_byte_offset, child_bitfield_bit_size,
+                child_bitfield_bit_offset, child_is_base_class,
+                child_is_deref_of_parent, valobj, language_flags))
+          return result;
+    }
+    // Clang types can be resolved even without a process.
+    bool is_signed;
+    if (CompilerType clang_type = GetAsClangTypeOrNull(type)) {
+      if (clang_type.IsEnumerationType(is_signed) && idx == 0)
+        // C enums get imported into Swift as structs with a "rawValue" field.
+        if (auto *ts =
+                llvm::dyn_cast<TypeSystemClang>(clang_type.GetTypeSystem()))
+          if (clang::EnumDecl *enum_decl = ts->GetAsEnumDecl(clang_type)) {
+            swift::Demangle::Demangler dem;
+            CompilerType raw_value =
+                CompilerType(ts, enum_decl->getIntegerType().getAsOpaquePtr());
+            child_name = "rawValue";
+            auto bit_size = raw_value.GetBitSize(
+                exe_ctx ? exe_ctx->GetBestExecutionContextScope() : nullptr);
+            child_byte_size = bit_size.getValueOr(0) / 8;
+            child_byte_offset = 0;
+            child_bitfield_bit_size = 0;
+            child_bitfield_bit_offset = 0;
+            child_is_base_class = false;
+            child_is_deref_of_parent = false;
+            language_flags = 0;
+            return RemangleAsType(dem, GetClangTypeTypeNode(dem, raw_value));
+          }
+      // Otherwise defer to TypeSystemClang.
+      if (CompilerType clang_child_type =
+              clang_type.GetChildCompilerTypeAtIndex(
+                  exe_ctx, idx, transparent_pointers, omit_empty_base_classes,
+                  ignore_array_bounds, child_name, child_byte_size,
+                  child_byte_offset, child_bitfield_bit_size,
+                  child_bitfield_bit_offset, child_is_base_class,
+                  child_is_deref_of_parent, valobj, language_flags)) {
+        swift::Demangle::Demangler dem;
+        return RemangleAsType(dem, GetClangTypeTypeNode(dem, clang_child_type));
+      }
+    }
+    if (!exe_scope)
+      LLDB_LOGF(GetLogIfAllCategoriesSet(LIBLLDB_LOG_TYPES),
+                "Cannot compute the children of type %s without an execution "
+                "context.",
```

Comment:
When would an execution context be missing? As a former consumer of these logs, I often wished for more detail. This seems like a message that could benefit from some explanation of how/why?